src/components/TopBar.js

```diff
@@ -2,27 +2,31 @@ import React from 'react';
 import styled from 'styled-components';
 import Button from './Button';
 import { connect } from 'react-redux';
-import { login, logout } from './../actions/auth.actions';
 import Text from './Text';
 import Colors from '../styles/Colors';
 import PropTypes from 'prop-types';
-
+import { requestLogin, requestLogout } from '../actions/auth.actions';
+import { ActivityIndicator } from '../shared/components/ActivityIndicator';
+//TODO make some AppError model and HelloGHError model (extends AppError)
 class TopBar extends React.Component {
     
     onClick = () => {
-        const { loggedUser, login, logout } = this.props;
-        const onPressAction = loggedUser ? logout : login;
+        const { logged, requestLogout, requestLogin } = this.props;
+        const onPressAction = logged ? requestLogout : requestLogin;
         onPressAction();
     };
     
     render() {
-        const { loggedUser } = this.props;
-        const buttonText = loggedUser ? 'Wyloguj' : 'Zaloguj';
+        const { logged, pending, user, error } = this.props;
+        const buttonText = logged ? 'Wyloguj' : 'Zaloguj';
         return (
             <TopBarWrapper>
                 <Text type="basic" color={ Colors.white } display={ 'inline' }>
-                    Zaloguj się przez GitHub
+                    {!logged && 'Zaloguj się'}
+                    {logged && user && user.email}
                 </Text>
+                {pending && <ActivityIndicator/>}
+                {error && <p>Błąd: {JSON.stringify(error)}</p>}
```

Comment:
Today we agreed on a meeting with @piecioshka that we're going to have an English version. I'm afraid it's good to start with i18n as soon as possible